src/ImageSharp/Formats/Png/Filters/AverageFilter.cs

```diff
@@ -79,6 +84,89 @@ public static void Encode(Span<byte> scanline, Span<byte> previousScanline, Span
                 sum += Numerics.Abs(unchecked((sbyte)res));
             }
 
+#if SUPPORTS_RUNTIME_INTRINSICS
+            if (Avx2.IsSupported)
+            {
+                Vector256<int> sumAccumulator = Vector256<int>.Zero;
+
+                for (int xLeft = x - bytesPerPixel; x + Vector256<byte>.Count <= scanline.Length; xLeft += Vector256<byte>.Count)
+                {
+                    Vector256<byte> scan = Unsafe.As<byte, Vector256<byte>>(ref Unsafe.Add(ref scanBaseRef, x));
+                    Vector256<byte> left = Unsafe.As<byte, Vector256<byte>>(ref Unsafe.Add(ref scanBaseRef, xLeft));
+                    Vector256<byte> above = Unsafe.As<byte, Vector256<byte>>(ref Unsafe.Add(ref prevBaseRef, x));
+
+                    Vector256<byte> res = Avx2.Subtract(scan, Average(left, above));
+                    Unsafe.As<byte, Vector256<byte>>(ref Unsafe.Add(ref resultBaseRef, x + 1)) = res; // +1 to skip filter type
+                    x += Vector256<byte>.Count;
+
+                    Vector256<sbyte> absRes = Avx2.Abs(res.AsSByte()).AsSByte();
+                    Vector256<short> loRes16 = Avx2.UnpackLow(absRes, Vector256<sbyte>.Zero).AsInt16();
+                    Vector256<short> hiRes16 = Avx2.UnpackHigh(absRes, Vector256<sbyte>.Zero).AsInt16();
+
+                    Vector256<int> loRes32 = Avx2.UnpackLow(loRes16, Vector256<short>.Zero).AsInt32();
+                    Vector256<int> hiRes32 = Avx2.UnpackHigh(loRes16, Vector256<short>.Zero).AsInt32();
+                    sumAccumulator = Avx2.Add(sumAccumulator, loRes32);
+                    sumAccumulator = Avx2.Add(sumAccumulator, hiRes32);
+
+                    loRes32 = Avx2.UnpackLow(hiRes16, Vector256<short>.Zero).AsInt32();
+                    hiRes32 = Avx2.UnpackHigh(hiRes16, Vector256<short>.Zero).AsInt32();
+                    sumAccumulator = Avx2.Add(sumAccumulator, loRes32);
+                    sumAccumulator = Avx2.Add(sumAccumulator, hiRes32);
+                }
+
+                for (int i = 0; i < Vector256<int>.Count; i++)
+                {
+                    sum += sumAccumulator.GetElement(i);
+                }
+            }
+            else if (Sse2.IsSupported)
+            {
+                var allBitsSet = Vector128.Create((sbyte)-1);
+                Vector128<int> sumAccumulator = Vector128<int>.Zero;
```

Comment:
.NET 5 produces optimal code for this. Older runtimes need a memory access for `allBitsSet` to load the "constant vector". If it's important to support these too, then
```suggestion
                Vector128<int> sumAccumulator = Vector128<int>.Zero;
                Vector128<sbyte> allBitsSet = Sse2.CompareEqual(sumAccumulator, sumAccumulator).AsSByte();

```